src/pluginscript_callbacks.lua

```diff
@@ -97,6 +98,14 @@ pluginscript_callbacks.script_init = wrap_callback(function(manifest, path, sour
 	manifest = ffi_cast('godot_pluginscript_script_manifest *', manifest)
 	path = tostring(ffi_cast('godot_string *', path))
 	source = tostring(ffi_cast('godot_string *', source))
+    local is_yue = path:sub(-4) == '.yue' 
+
+    if is_yue then
+        local yue = require('yue')
+        local codes, err, globals = yue.to_lua(source)
```

Comment:
If the file is a `.yue` file it loads the YueScript shared library (which we can compile for our user or ask our user to obtain themselves).
It then compiles the source to be Lua code. The rest of the code should be compatible from this point

Reply:
We'd better compile and/or distribute the Yuescript shared library to end users, since it will be needed in runtime for the game/app. If the code would be compiled ahead-of-time as part of the build process, for example, than it would be more ok (although still somewhat annoying) to ask devs for obtaining it by themselves.